robot_smach_states/src/robot_smach_states/utility.py

```diff
@@ -351,6 +353,109 @@ def execute(self, userdata=None):
         return "done"
 
 
+class ResolveArm(smach.State):
```

Comment:
I really don't get why we have a state to resolve the arm after which we then use userdata to pass on the arm.

So why do we need to have a state to resolve the arm? Why can't this happen in a designator which is resolved by the first state that uses it?

Why do we use usedata after resolving the arm instead of a designator?

Reply:
With the smach state system we do need a state before navigate_to_grasp to put the arm into the userdata. That is a requirement.
As for userdata over designator. I remember some discussion between people smarter than me about this